.gitlab/test.yml

```diff
@@ -0,0 +1,33 @@
+test:
+  extends: .check_skip_rust
+  timeout: 2 hours
+  image: paritytech/ci-unified:bullseye-1.77.0
+  stage: test
+  cache:
+    key: cache-test
+    paths:
+      - target/
+      - $HOME/.cargo/bin/
+      - $HOME/.cargo/registry/index/
+      - $HOME/.cargo/registry/cache/
+      - $HOME/.cargo/git/db/
+  script:
+    - time cargo test --all --all-targets --locked
+
+test-features:
+  extends: .check_skip_rust
+  timeout: 2 hours
+  image: paritytech/ci-unified:bullseye-1.77.0
+  stage: test
+  needs:
+    - test
```

Comment:
There is not really a dependency between this and the test step, so we could run these two in parallel, as long as they have a different cache space.